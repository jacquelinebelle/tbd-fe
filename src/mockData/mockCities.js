const mockCities = [
  {
    city: "Mesa, AZ",
    mobile:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/phoenix-9fedc88f57.jpg",
    web:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/phoenix_web-92224752db.jpg",
    teleport_city_score: 56.66256756756756,
    summary:
      "<p>Phoenix, Arizona, is among the top cities with a <b>free business environment</b>. According to our city rankings, this is a good place to live with high ratings in <b>healthcare</b>, <b>leisure & culture</b> and <b>outdoors</b>. </p> <p>Phoenix is one of the top ten city matches for 0.2% of Teleport users.</p> ",
    categories: [
      {
        color: "#f3c32c",
        name: "Housing",
        score_out_of_10: 6.5329999999999995
      },
      { color: "#f3d630", name: "Cost of Living", score_out_of_10: 6.208 },
      {
        color: "#f4eb33",
        name: "Startups",
        score_out_of_10: 6.508500000000001
      },
      { color: "#d2ed31", name: "Venture Capital", score_out_of_10: 4.558 },
      {
        color: "#7adc29",
        name: "Travel Connectivity",
        score_out_of_10: 3.3024999999999998
      },
      { color: "#36cc24", name: "Commute", score_out_of_10: 4.828500000000001 },
      { color: "#19ad51", name: "Business Freedom", score_out_of_10: 8.671 },
      { color: "#0d6999", name: "Safety", score_out_of_10: 2.5829999999999997 },
      {
        color: "#051fa5",
        name: "Healthcare",
        score_out_of_10: 8.567333333333334
      },
      {
        color: "#150e78",
        name: "Education",
        score_out_of_10: 4.590999999999999
      },
      {
        color: "#3d14a4",
        name: "Environmental Quality",
        score_out_of_10: 4.625249999999999
      },
      { color: "#5c14a1", name: "Economy", score_out_of_10: 6.5145 },
      { color: "#88149f", name: "Taxation", score_out_of_10: 4.488 },
      { color: "#b9117d", name: "Internet Access", score_out_of_10: 6.178 },
      {
        color: "#d10d54",
        name: "Leisure & Culture",
        score_out_of_10: 6.8100000000000005
      },
      {
        color: "#e70c26",
        name: "Tolerance",
        score_out_of_10: 5.5649999999999995
      },
      { color: "#f1351b", name: "Outdoors", score_out_of_10: 6.7355 }
    ]
  },
  {
    city: "Durham, NC",
    mobile:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/raleigh-e74249142f.jpg",
    web:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/raleigh_web-2909c09abe.jpg",
    teleport_city_score: 62.51567567567569,
    summary:
      "<p>Raleigh, North Carolina, is among the top citie…ten city matches for 3.4% of Teleport users.</p>",
    categories: [
      { color: "#f3c32c", name: "Housing", score_out_of_10: 6.128500000000001 },
      {
        color: "#f3d630",
        name: "Cost of Living",
        score_out_of_10: 5.610000000000001
      },
      { color: "#f4eb33", name: "Startups", score_out_of_10: 5.497 },
      {
        color: "#d2ed31",
        name: "Venture Capital",
        score_out_of_10: 5.157000000000002
      },
      {
        color: "#7adc29",
        name: "Travel Connectivity",
        score_out_of_10: 1.7570000000000001
      },
      { color: "#36cc24", name: "Commute", score_out_of_10: 4.78425 },
      { color: "#19ad51", name: "Business Freedom", score_out_of_10: 8.671 },
      { color: "#0d6999", name: "Safety", score_out_of_10: 6.5225 },
      {
        color: "#051fa5",
        name: "Healthcare",
        score_out_of_10: 8.966666666666667
      },
      { color: "#150e78", name: "Education", score_out_of_10: 7.1165 },
      {
        color: "#3d14a4",
        name: "Environmental Quality",
        score_out_of_10: 8.583750000000002
      },
      { color: "#5c14a1", name: "Economy", score_out_of_10: 6.5145 },
      { color: "#88149f", name: "Taxation", score_out_of_10: 4.062 },
      { color: "#b9117d", name: "Internet Access", score_out_of_10: 7.9515 },
      {
        color: "#d10d54",
        name: "Leisure & Culture",
        score_out_of_10: 6.6129999999999995
      },
      {
        color: "#e70c26",
        name: "Tolerance",
        score_out_of_10: 7.562000000000001
      },
      {
        color: "#f1351b",
        name: "Outdoors",
        score_out_of_10: 3.3905000000000003
      }
    ]
  },
  {
    city: "Minneapolis, MN",
    mobile:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/minneapolis-saint-paul-0167dd447d.jpg",
    web:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/minneapolis-saint-paul_web-16dbfd4210.jpg",
    teleport_city_score: 61.163648648648646,
    summary:
      "<p>Minneapolis-Saint Paul, Minnesota, is among the…ten city matches for 4.5% of Teleport users.</p>↵",
    categories: [
      { color: "#f3c32c", name: "Housing", score_out_of_10: 4.738500000000001 },
      {
        color: "#f3d630",
        name: "Cost of Living",
        score_out_of_10: 5.042000000000001
      },
      {
        color: "#f4eb33",
        name: "Startups",
        score_out_of_10: 6.5264999999999995
      },
      {
        color: "#d2ed31",
        name: "Venture Capital",
        score_out_of_10: 5.5230000000000015
      },
      { color: "#7adc29", name: "Travel Connectivity", score_out_of_10: 3.544 },
      { color: "#36cc24", name: "Commute", score_out_of_10: 4.924999999999999 },
      { color: "#19ad51", name: "Business Freedom", score_out_of_10: 8.671 },
      { color: "#0d6999", name: "Safety", score_out_of_10: 5.085000000000001 },
      { color: "#051fa5", name: "Healthcare", score_out_of_10: 8.972 },
      { color: "#150e78", name: "Education", score_out_of_10: 5.2745 },
      {
        color: "#3d14a4",
        name: "Environmental Quality",
        score_out_of_10: 8.362
      },
      { color: "#5c14a1", name: "Economy", score_out_of_10: 6.5145 },
      { color: "#88149f", name: "Taxation", score_out_of_10: 4.062 },
      {
        color: "#b9117d",
        name: "Internet Access",
        score_out_of_10: 6.258499999999999
      },
      { color: "#d10d54", name: "Leisure & Culture", score_out_of_10: 6.8505 },
      { color: "#e70c26", name: "Tolerance", score_out_of_10: 7.288 },
      { color: "#f1351b", name: "Outdoors", score_out_of_10: 5.095999999999999 }
    ]
  },
  {
    city: "Calabasas, CA",
    mobile:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/los-angeles-f5b60deb04.jpg",
    web:
      "https://d13k13wj6adfdf.cloudfront.net/urban_areas/los-angeles_web-2eedb2a83e.jpg",
    teleport_city_score: 63.426351351351364,
    summary:
      "<p>↵    Los Angeles is a sprawling city of 500 squ…en city matches for 11.5% of Teleport users.</p>↵",
    categories: [
      {
        color: "#f3c32c",
        name: "Housing",
        score_out_of_10: 1.5274999999999999
      },
      {
        color: "#f3d630",
        name: "Cost of Living",
        score_out_of_10: 4.556000000000001
      },
      { color: "#f4eb33", name: "Startups", score_out_of_10: 10 },
      { color: "#d2ed31", name: "Venture Capital", score_out_of_10: 10 },
      {
        color: "#7adc29",
        name: "Travel Connectivity",
        score_out_of_10: 3.9585000000000004
      },
      {
        color: "#36cc24",
        name: "Commute",
        score_out_of_10: 3.6770000000000005
      },
      {
        color: "#19ad51",
        name: "Business Freedom",
        score_out_of_10: 8.574666666666667
      },
      { color: "#0d6999", name: "Safety", score_out_of_10: 5.705 },
      {
        color: "#051fa5",
        name: "Healthcare",
        score_out_of_10: 8.439666666666666
      },
      { color: "#150e78", name: "Education", score_out_of_10: 8.6245 },
      {
        color: "#3d14a4",
        name: "Environmental Quality",
        score_out_of_10: 4.7315000000000005
      },
      { color: "#5c14a1", name: "Economy", score_out_of_10: 6.5145 },
      { color: "#88149f", name: "Taxation", score_out_of_10: 4.7675 },
      {
        color: "#b9117d",
        name: "Internet Access",
        score_out_of_10: 5.496500000000001
      },
      { color: "#d10d54", name: "Leisure & Culture", score_out_of_10: 9.196 },
      {
        color: "#e70c26",
        name: "Tolerance",
        score_out_of_10: 6.959999999999999
      },
      { color: "#f1351b", name: "Outdoors", score_out_of_10: 6.747 }
    ]
  }
];

export default mockCities;
