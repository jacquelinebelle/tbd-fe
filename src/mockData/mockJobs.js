const mockJobs = [
  {
    title: "JavaScript Developer",
    location: "Mesa, AZ",
    snippet: "&nbsp;...server side development using excellent, modern tools",
    salary: "",
    source: "Careerbuilder.com",
    type: "Full-time",
    link:
      "https://us.jooble.org/away/2489493276763186316?p=1&pos=1&cid=512&ckey=Javascript&age=134&relb=175&brelb=115&scr=7278,57184782609&bscr=4783,0615&aq=19401826",
    company: "IQVIA",
    updated: "2019-09-04T03:13:55.1458976+03:00",
    id: 2489493276763186000
  },
  {
    title: "JavaScript Developer",
    location: "Durham, NC",
    snippet: "&nbsp;...<b>JavaScript </b>Developer - Mid-Level to Senior",
    salary: "100000",
    source: "Careerbuilder.com",
    type: "Full-time",
    link:
      "https://us.jooble.org/away/-2476942630518357134?p=1&pos=2&cid=512&ckey=Javascript&age=105&relb=175&brelb=115&scr=7240,01630434783&bscr=4757,725&aq=19401826",
    company: "Vaco Technology",
    updated: "2019-09-05T08:01:17.9188044+03:00",
    id: -2476942630518357000
  },
  {
    title: "Javascript Developer",
    location: "Minneapolis, MN",
    snippet: "&nbsp;...who will welcome your talents, as well as your guts",
    salary: "",
    source: "ziprecruiter.com",
    type: "",
    link:
      "https://us.jooble.org/away/-7666096497938369673?p=1&pos=5&cid=512&ckey=Javascript&age=137&relb=168&brelb=115&scr=7037,61787826087&bscr=4817,417&aq=19401826",
    company: "REDCIRCLE LLC",
    updated: "2019-09-04T00:00:00.0000000",
    id: -7666096497938370000
  },
  {
    title: "Javascript Developer",
    location: "Calabasas, CA",
    snippet:
      "We are looking for a <b>JavaScript </b>Developer who is motivated",
    salary: "",
    source: "ziprecruiter.com",
    type: "",
    link:
      "https://us.jooble.org/away/3689689097643763247?p=1&pos=6&cid=512&ckey=Javascript&age=113&relb=168&brelb=115&scr=7013,53982608696&bscr=4800,935&aq=19401826",
    company: "Quantum Solutions",
    updated: "2019-09-05T00:00:00.0000000",
    id: 3689689097643763000
  }
];

export default mockJobs;
