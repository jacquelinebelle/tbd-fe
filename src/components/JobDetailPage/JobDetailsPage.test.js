import React from "react";
import { shallow } from "enzyme";
import { JobDetailPage, mapDispatchToProps, mapStateToProps } from "./index";
import mockJobs from '../../mockData/mockJobs';
import mockCities from '../../mockData/mockCities';

const mockCity = mockCities[0]

describe("Job Details Page", () => {
  let wrapper;
  let mockState;

  beforeEach(function(){
    mockState = {
      jobs: mockJobs,
      currentCity: mockCity
    };
    wrapper = shallow(
      <JobDetailPage jobs={mockJobs} 
      currentCity={mockCity}
      id={2489493276763186000}/>
    )
  })
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot', () => {
    const mockState = {
      currJob: {},
      cityDetails: [],
      scores: false,
      details: false,
      showSalaries:false,
      shownDetail: ''
  }
    expect(wrapper.state()).toMatchSnapshot(mockState);
  })

  it('should update state with handleState on click of section', () => {
    wrapper.find('.detail-city-more').at(0).simulate('click')
    setTimeout(function(){ expect(wrapper.state().scores).toEqual(true) }, 1000);
  })

  it('should update state with handleState on click of section', () => {
    wrapper.find('.detail-city-more').at(1).simulate('click')
    setTimeout(function(){ expect(wrapper.state().showSalaries).toEqual(true) }, 1000);
  })

  it('should update state with handleState on click of section', () => {
    wrapper.find('.detail-title').at(2).simulate('click')
    setTimeout(function(){ expect(wrapper.state().details).toEqual(true) }, 1000);
  })

  describe("mapDispatchToProps", () => {
    it("should call scoreThunk when dispatch", () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.cityThunk(mockCity);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });

  describe("mapStateToProps", () => {
    it("should reflect whats in the redux store", () => {
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
    });
  });
})