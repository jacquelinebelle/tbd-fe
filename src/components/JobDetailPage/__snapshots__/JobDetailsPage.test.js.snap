// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Details Page should match snapshot 1`] = `
<article
  className="job-detail"
>
  <img
    className="detail-img"
    src="https://d13k13wj6adfdf.cloudfront.net/urban_areas/phoenix_web-92224752db.jpg"
  />
  <div
    className="detail-sections-container"
  >
    <section
      className="job-details-section"
    >
      <h3
        className="details-job-title"
      />
      <a>
        Link to the listing.
      </a>
      <p
        className="detail-job-salary"
      />
      <p
        className="detail-snippet"
      />
      <p
        className="detail-job-update"
      />
    </section>
    <section
      className="city-details-section"
    >
      <h3 />
      <p
        className="detail-city-disclaimer"
      >
        Based on nearest major city.
      </p>
      <p
        className="detail-city-summary"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Phoenix, Arizona, is among the top cities with a <b>free business environment</b>. According to our city rankings, this is a good place to live with high ratings in <b>healthcare</b>, <b>leisure & culture</b> and <b>outdoors</b>. </p> <p>Phoenix is one of the top ten city matches for 0.2% of Teleport users.</p> ",
          }
        }
      />
      <p
        className="detail-teleport-score"
      >
        Overal Teleport Score: 56.66
      </p>
    </section>
  </div>
  <div
    className="false detail-city-more"
    onClick={[Function]}
  >
    <h4
      className="detail-title"
    >
      All Scores
    </h4>
    <section
      className="city-scores"
    />
  </div>
  <div
    className=" detail-city-more"
    onClick={[Function]}
  >
    <h4
      className="detail-title"
    >
      All Details
    </h4>
    <section
      className="city-details"
    />
  </div>
</article>
`;
